### PR TITLE
Update blip model card

### PR DESCRIPTION
**Pull request by devkade**
# What does this PR do?

I have changed the documentation of the [blip](https://huggingface.co/docs/transformers/v4.52.1/en/model_doc/blip) model according to the following issue https://github.com/huggingface/transformers/issues/36979#issue-2947704577.

- [x] Standardize model card with a consistent format
- [x] Provide code examples featuring the `Pipeline`, `AutoModel`(not `transformers-cli`).
- [x] Attention mask visualizer supporting --> It seems not to be supported in blip.

Because it is the first contribution, it may be insufficient. If correction or supplementation is needed, please feel free to comment at any time.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @stevhliu

File: docs/source/en/model_doc/blip.md [hunk 1]
```diff
@@ -14,31 +14,78 @@ rendered properly in your Markdown viewer.
...
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
-</div>
+[BLIP](https://huggingface.co/papers/2201.12086) (Bootstrapped Language-Image Pretraining) is a multimodal model that can understand and generate images and text together, and it is distinctive in that one model can handle various vision-language tasks such as image captioning, image-text matching, and VQA all together. Unlike existing VLP models that were specialized in only one of understanding or generation, BLIP is designed to be flexibly transferred to both domains. In particular, without using noisy image-text pairs collected from the web as they are, it introduces a ‘bootstrapping caption’ technique that generates sentences with its own captioner and goes through a filtering process to increase learning quality, thereby securing cleaner and more meaningful data. As a result, BLIP achieves high performance and excellent generalization despite little artificial manual labor, and can be effectively utilized for various multimodal tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/blip.md]:
```suggestion
[BLIP](https://huggingface.co/papers/2201.12086) (Bootstrapped Language-Image Pretraining) is a vision-language pretraining (VLP) framework designed for *both* understanding and generation tasks. Most existing pretrained models are only good at one or the other. It uses a captioner to generate captions and a filter to remove the noisy captions. This increases training data quality and more effectively uses the messy web data.
```

File: docs/source/en/model_doc/blip.md [hunk 2]
```diff
@@ -14,31 +14,78 @@ rendered properly in your Markdown viewer.
...
+
+
+You can find all the original BLIP checkpoints under the [BLIP](https://huggingface.co/collections/Salesforce/blip-models-65242f40f1491fbf6a9e9472) collection.
+
+> [!TIP]
+> Click on the BLIP models in the right sidebar for more examples of how to apply BLIP to different tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/blip.md]:
```suggestion
> Click on the BLIP models in the right sidebar for more examples of how to apply BLIP to different vision language tasks.
```

File: docs/source/en/model_doc/blip.md [hunk 3]
```diff
@@ -14,31 +14,78 @@ rendered properly in your Markdown viewer.
...
+
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/blip.md]:
```suggestion
The example below demonstrates how to visual question answering with [`Pipeline`] or the [`AutoModel`] class.
```

File: docs/source/en/model_doc/blip.md [hunk 4]
```diff
@@ -14,31 +14,78 @@ rendered properly in your Markdown viewer.
...
 
-## Overview
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-The BLIP model was proposed in [BLIP: Bootstrapping Language-Image Pre-training for Unified Vision-Language Understanding and Generation](https://arxiv.org/abs/2201.12086) by Junnan Li, Dongxu Li, Caiming Xiong, Steven Hoi.
+```python
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/blip.md:42]: Lets use a more interesting image!

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="visual-question-answering",
    model="Salesforce/blip-vqa-base",
    torch_dtype=torch.float16,
    device=0
)
url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
pipeline(question="What is the weather in this image?", image=url)
```

File: docs/source/en/model_doc/blip.md [hunk 5]
```diff
@@ -14,31 +14,78 @@ rendered properly in your Markdown viewer.
...
+    torch_dtype=torch.float16,
+    device=0
+)
+url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+print(pipeline(question="What is cat doing?", image=url))
+```
 
-The abstract from the paper is the following:
+</hfoption>
+<hfoption id="AutoModel">
```
Comment (stevhliu) [inline docs/source/en/model_doc/blip.md:56]: Same with the `AutoModel` example.

```py
import requests
import torch
from PIL import Image
from transformers import AutoProcessor, AutoModelForVisualQuestionAnswering

processor = AutoProcessor.from_pretrained("Salesforce/blip-vqa-base")
model = AutoModelForVisualQuestionAnswering.from_pretrained(
    "Salesforce/blip-vqa-base", 
    torch_dtype=torch.float16,
    device_map="auto"
)

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = Image.open(requests.get(url, stream=True).raw)

question = "What is the weather in this image?"
inputs = processor(images=image, text=question, return_tensors="pt").to("cuda", torch.float16)

output = model.generate(**inputs)
processor.batch_decode(output, skip_special_tokens=True)[0]
```

File: docs/source/en/model_doc/blip.md [hunk 6]
```diff
@@ -14,31 +14,78 @@ rendered properly in your Markdown viewer.
...
+</hfoptions>
 
 ![BLIP.gif](https://cdn-uploads.huggingface.co/production/uploads/1670928184033-62441d1d9fdefb55a0b7d12c.gif)
 
-This model was contributed by [ybelkada](https://huggingface.co/ybelkada).
-The original code can be found [here](https://github.com/salesforce/BLIP).
+## Notes
+
+- This model was contributed by [ybelkada](https://huggingface.co/ybelkada).
+- The original code can be found [here](https://github.com/salesforce/BLIP).
```
Comment (stevhliu) [inline docs/source/en/model_doc/blip.md]:
```suggestion
```

File: docs/source/en/model_doc/blip.md [hunk 7]
```diff
@@ -14,31 +14,78 @@ rendered properly in your Markdown viewer.
...
 
 ## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/blip.md:85]: For the notebook below, lets use this language:

```md
Refer to this [notebook](https://github.com/huggingface/notebooks/blob/main/examples/image_captioning_blip.ipynb) to learn how to fine-tune BLIP for image captioning on a custom dataset.
```

**Review by stevhliu — COMMENTED** [review]
Very nice job! I made some suggestions to simplify things a bit :)

**Comment by devkade** [comment]
Thank you for the feedback. The revision was a little late. I have incorporated all of the suggested revisions, and if there are any additional revisions, please let me know anytime.

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution! 🤗
